### PR TITLE
MULTISITE-15705 - Detect empty strongarm for pathauto.

### PR DESCRIPTION
**Pull request by jonhy81**

### CODE REVIEW

File: phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    $end = (count($phpcsFile->getTokens()) + 1);
+
+    // If the file extension is diferent from strongarm.inc return.
+    $fileName = $phpcsFile->getFilename();
+    $fileExtension = strtolower(substr($fileName, -13));
+    if ($fileExtension !== 'strongarm.inc') {
+      return $end;
+    }
+
+    // Check whole project for string "sites/".
```
Comment (iulianweb) [inline phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php]: You mean "pathauto" here right ? :p

File: phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ * PHP version 5
+ *
+ * @category PHP
+ * @package  PHP_CodeSniffer
+ * @link     http://pear.php.net/package/PHP_CodeSniffer
+ */
+
+/**
+ * Hardcoded paths to theme or modules need to be removed or replace by a call
```
Comment (iulianweb) [inline phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php]: Please fix to something like Strongarm settings shouldn't be delivered when the value is empty, especially with pathauto patterns.

File: phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    $token  = $tokens[$stackPtr];
+
+    // If our token have the keyword pathauto, check next property and value.
+    if (strpos($token['content'], 'pathauto') !== false) {
+      if (
+        $tokens[$stackPtr + 10]['content'] == "''" &&
+        $tokens[$stackPtr + 6]['content'] == 'value'
+      ) {
+        $error = "Empty strongarm settings for " . $token['content'] . " are not allowed.";
+        $phpcsFile->addError($error, $stackPtr, 'EmptySettings');
```
Comment (iulianweb) [inline phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php]: Maybe improve to have more clear message like error:QualityAssurance.Generic.EmptySettings.Strongarm ?

File: phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+   *
+   * @param PHP_CodeSniffer_File $phpcsFile The file being scanned.
+   * @param int                  $stackPtr  The position of the current token in the
+   *                                        stack passed in $tokens.
+   *
+   * @return void
+   */
+  public function process(PHP_CodeSniffer_File $phpcsFile, $stackPtr)
+  {
+    // Only run this sniff once per strongarm exported file file.
```
Comment (iulianweb) [inline phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php]: File mentioned twice.

**Review by iulianweb — CHANGES_REQUESTED** [review]

**Comment by jonhy81** [comment]
Thanks @iulianweb, can you test again?

File: phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php (hunk 2 above)
Comment (iulianweb) [reply phpcs/Standards/QualityAssurance/Sniffs/Generic/EmptySettingsSniff.php]: This one has not been fixed.

**Review by iulianweb — APPROVED** [review]

**Pull request merged**